File: Sorting/56.Merge_Intervals/Merge_Intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& arr) {
        vector<vector<int>> ans;
        sort(arr.begin(), arr.end());
        int arr_start = arr[0][0], arr_end = arr[0][1];
        for(int i = 1; i < size(arr); i++){
            if(arr[i][0] <= arr_end){
                if (arr[i][1] > arr_end){
                    arr_end = arr[i][1];
                }
            }
            else{
                ans.push_back({arr_start, arr_end});
                arr_start = arr[i][0];
                arr_end = arr[i][1];
            }
        }
        ans.push_back({arr_start, arr_end});
        return ans;
    }
};

File: Sorting/56.Merge_Intervals/Merge_Intervals.py
class Solution:
    def merge(self, arr: List[List[int]]) -> List[List[int]]:
        ans = []
        arr.sort()
        arr_start = arr[0][0]; arr_end = arr[0][1]
        for i in range(1, len(arr)):
            if arr[i][0] <= arr_end: 
                if arr[i][1] > arr_end:
                    arr_end = arr[i][1]
            else:
                ans.append([arr_start, arr_end])
                arr_start = arr[i][0]
                arr_end = arr[i][1]
        ans.append([arr_start, arr_end])
        return ans